src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1989,8 +1989,10 @@ private IEnumerable<Diagnostic> FreezeDeclarationDiagnostics()
         {
             _declarationDiagnosticsFrozen = true;
 
-            // Also freeze that flag, as symbols bound after getting the declaration diagnostics shouldn't need to modify it
-            _needsGeneratedIsReadOnlyAttribute_IsFrozen = true;
+            // Also freeze generated attribute flags by observing them
+            // symbols bound after getting the declaration diagnostics shouldn't need to modify the flags
+            _ = NeedsGeneratedIsReadOnlyAttribute;
+            _ = NeedsGeneratedIsByRefLikeAttribute;
```

Comment:
>_ = NeedsGeneratedIsByRefLikeAttribute; [](start = 12, length = 39)

This is very clever, but I think it is better to assign to ..._IsFrozen fields explicitly so that it is easier to navigate/understand the codebase.